Added User property to show if it has 2fa or not.

AntonCroner opened the pull request:
Removed unused complaining overload.
Fixed spelling mistake.

EliasEriksson commented on Context/Users/Storage/Context/Users/User.ts:
```diff
@@ -89,33 +90,14 @@ export namespace User {
+					password: await Password.hash(user.password.new, context.secret),
+					permissions: { [context.application]: flagly.parse(user.permissions) as userwidgets.User.Permissions },
 					modified: now,
 					created: now,
```
this change will cause the created property to be updated to now everytime. the fix for this would be to create two functions instead of one that is overloaded
suggestion:
```ts
export const from = {
    model: (...) => {...},
    creatable: (...) => {...},
}
```

AntonCroner replied:
But this function is only called once, at create time. maybe it should be called create?

EliasEriksson commented on Context/Users/Storage/Context/Users/User.ts:
```diff
@@ -34,6 +34,7 @@ export namespace User {
 		return (({ password, permissions, twoFactor, ...user }) => ({
 			...user,
 			permissions: flagly.Flags.stringify(permissions[context.application] ?? {}),
+			...("twoFactor" in user && user.twoFactor ? { twoFactor: true } : {}),
```
maybe set this explicitly to false?. with the changes we discussed in slack undefined could mean that we are not allowed to see it.

AntonCroner replied:
\+

EliasEriksson commented:
pages log:
```
<pre class="c_nr c_bv c_mu c_mv c_rs c_rt c_db c_sf c_sg c_sh c_si c_ey c_ez c_ru c_bx c_sj">
11:35:28.008 | Cloning repository...
-- | --
11:35:28.766 | From https://github.com/userwidgets/api
11:35:28.767 | * branch            af3903db485c88165b0cacc0c8a995b904a79a7a -> FETCH_HEAD
11:35:28.767 |  
11:35:28.801 | HEAD is now at af3903d Added User property to show if it has 2fa or not. Removed unused complaining overload. Fixed spelling mistake.
11:35:28.801 |  
11:35:28.895 |  
11:35:28.921 | Success: Finished cloning repository files
11:35:29.687 | Installing dependencies
11:35:29.694 | Python version set to 2.7
11:35:32.419 | Downloading and installing node v16.13.0...
11:35:32.837 | Downloading https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.xz...
11:38:07.276 | curl: (18) transfer closed with 8990548 bytes remaining to read
11:38:07.291 | Binary download from https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.xz failed, trying source.
11:38:07.298 | grep: /opt/buildhome/.nvm/.cache/bin/node-v16.13.0-linux-x64/node-v16.13.0-linux-x64.tar.xz: No such file or directory
11:38:07.299 | Provided file to checksum does not exist.
11:38:07.316 | Binary download failed, trying source.
11:38:07.341 | Detected that you have 8 CPU core(s)
11:38:07.341 | Running with 7 threads to speed up the build
11:38:07.532 | clang: error while loading shared libraries: libz3.so.4: cannot open shared object file: No such file or directory
11:38:07.795 | Downloading https://nodejs.org/dist/v16.13.0/node-v16.13.0.tar.xz...
11:39:57.276 | curl: (18) transfer closed with 31651696 bytes remaining to read
11:39:57.288 | Binary download from https://nodejs.org/dist/v16.13.0/node-v16.13.0.tar.xz failed, trying source.
11:39:57.294 | grep: /opt/buildhome/.nvm/.cache/src/node-v16.13.0/node-v16.13.0.tar.xz: No such file or directory
11:39:57.294 | Provided file to checksum does not exist.
11:39:57.305 | Failed to install node version '16.13.0'
11:39:57.312 | Failed: build command exited with code: 1
11:39:58.095 | Failed: error occurred while running build command

</pre>
```
set node version to 20

AntonCroner commented:
> pages log:
> 
> ```
> 11:35:28.008 | Cloning repository...
> -- | --
> 11:35:28.766 | From https://github.com/userwidgets/api
> 11:35:28.767 | * branch            af3903db485c88165b0cacc0c8a995b904a79a7a -> FETCH_HEAD
> 11:35:28.767 |  
> 11:35:28.801 | HEAD is now at af3903d Added User property to show if it has 2fa or not. Removed unused complaining overload. Fixed spelling mistake.
> 11:35:28.801 |  
> 11:35:28.895 |  
> 11:35:28.921 | Success: Finished cloning repository files
> 11:35:29.687 | Installing dependencies
> 11:35:29.694 | Python version set to 2.7
> 11:35:32.419 | Downloading and installing node v16.13.0...
> 11:35:32.837 | Downloading https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.xz...
> 11:38:07.276 | curl: (18) transfer closed with 8990548 bytes remaining to read
> 11:38:07.291 | Binary download from https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.xz failed, trying source.
> 11:38:07.298 | grep: /opt/buildhome/.nvm/.cache/bin/node-v16.13.0-linux-x64/node-v16.13.0-linux-x64.tar.xz: No such file or directory
> 11:38:07.299 | Provided file to checksum does not exist.
> 11:38:07.316 | Binary download failed, trying source.
> 11:38:07.341 | Detected that you have 8 CPU core(s)
> 11:38:07.341 | Running with 7 threads to speed up the build
> 11:38:07.532 | clang: error while loading shared libraries: libz3.so.4: cannot open shared object file: No such file or directory
> 11:38:07.795 | Downloading https://nodejs.org/dist/v16.13.0/node-v16.13.0.tar.xz...
> 11:39:57.276 | curl: (18) transfer closed with 31651696 bytes remaining to read
> 11:39:57.288 | Binary download from https://nodejs.org/dist/v16.13.0/node-v16.13.0.tar.xz failed, trying source.
> 11:39:57.294 | grep: /opt/buildhome/.nvm/.cache/src/node-v16.13.0/node-v16.13.0.tar.xz: No such file or directory
> 11:39:57.294 | Provided file to checksum does not exist.
> 11:39:57.305 | Failed to install node version '16.13.0'
> 11:39:57.312 | Failed: build command exited with code: 1
> 11:39:58.095 | Failed: error occurred while running build command
> ```
> 
> set node version to 20

All the workflows are set to node 20, i think this is inside the pages settings on cloudflare dashboard.
